src/libaudcore/mainloop.cc

```diff
@@ -224,13 +224,27 @@ class QueuedFuncTimer : public QueuedFuncHelper, public QObject
     }
 
     void stop ()
-        { deleteLater (); }
+    {
+        if (timerId)
+        {
+            killTimer(timerId);
```

Comment:
We're in a different thread here than the timer is running in, so even if killTimer() is safe (which is questionable), it's no better than deleteLater() anyway.